### PR TITLE
Flake8 5 and final

### PR DESCRIPTION
**Pull request by bobhy**
Fifth and assuredly final of 3 PRs planned to resolve accumulated ad-hoc lint exceptions in xonsh codebase.  Refer to `setup.cfg [flake8]` to evaluate progress towards this end. 

This PR focused on resolving F401 (unused import) warnings, but it includes all the changes from the previous work, so could be reviewed relative to master and merged on its own.

Alternatively, review #3561, then #3560, #3580 and #3591 and merge those before looking at this one, then it will be a more modest diff set.

### CODE REVIEW

**Comment by bobhy** [comment]
@gforsyth Could you take a look at this and the preceding PRs?
I know this one is no less formidable than the one you asked me to break up, but there is a reasonable sequence to it:  #3561, then #3560, #3580 and #3591 and finally this one..

File: xonsh/completers/python.py [hunk 1]
```diff
@@ -206,11 +206,11 @@ def _safe_eval(expr, ctx):
     try:
         val = xonsh_safe_eval(expr, ctx, ctx, transform=False)
         _ctx = ctx
-    except:  # pylint:disable=bare-except
```
Comment (scopatz) [inline xonsh/completers/python.py:209]: I think that these were bare-excepts for a good reason.  We don't want a Ctrl+C or a raise SystemExit in a completer to take down xonsh

File: xonsh/inspectors.py [hunk 2]
```diff
@@ -288,7 +288,7 @@ def find_file(obj):
             except TypeError:
                 # Can happen for builtins
                 pass
-    except:  # pylint:disable=bare-except
```
Comment (scopatz) [inline xonsh/inspectors.py:291]: Same here

File: xonsh/xoreutils/which.py [hunk 3]
```diff
@@ -87,7 +87,7 @@ def print_path(abs_name, from_where, stdout, verbose=False, captured=False):
         p, f = os.path.split(abs_name)
         f = next(s.name for s in xp.scandir(p) if s.name.lower() == f.lower())
         abs_name = os.path.join(p, f)
-        if builtins.__xonsh__.env.get("FORCE_POSIX_PATHS", False):
```
Comment (scopatz) [inline xonsh/xoreutils/which.py:90]: I am unclear why the `builtins` module is being removed

**Comment by bobhy** [comment]
I'm going to withdraw this stale and unfocused PR and refactor the work into a set of more comprehensible PRs that will hopefully be more acceptable, or at least falsifiable.

**Pull request closed without merging**